File: WebSocketChat.Frontend/src/app/shared/models/chat/chat-room.ts
import { Subject } from 'rxjs/Rx';
import { EventEmitter } from '@angular/core';

import { UserJoinMessage } from './messages/user-join-message';
import { ChatUser } from './chat-user';
import { NicknameRequest } from './messages/nickname-request';
import { UserLeaveMessage } from './messages/user-leave-message';
import { ChatMessage } from './messages/chat-message';

export class ChatRoom {
	/** Identifier of the room. The number 0 indicates the global server room. */
	roomIdentifier: number;

	/** Name of the chatroom */
	name: string;

	/** Description of the chatroom */
	description: string;	

	/** Indicates whether this is a private room with another user  */
	isPrivateRoom : boolean;

	/** Identifier-GUIDs of all users which have joined this room */
	connectedUsers: string[] = [];

	/** Indicates whether the room is password protected */
	isPasswordProtected: boolean;

	/** Base64 string representation of the room's avatar */
	roomAvatarBase64: string = "";

	/** Client-side only properties which will not be send by the server */

	/** We will save all received messages in this array (client-side only) */
	messages: Array<ChatMessage | UserJoinMessage | UserLeaveMessage | NicknameRequest> = [];

	/** The timestamp of the last message in this room */
	lastMessageTimestamp: Date;

	/** We will resolve all user-guids to actual chat user objects */
	connectedUsersResolved : Array<ChatUser> = [];

	/** Concatenated usernames of all users besides yourself */
	nicknameStringAggregation: string;

	/** Specifies whether the user has already joined the room */
	hasJoinedRoom: boolean;

	/** If the user fails to type in the correct password to join the room, we will fire this event so that the UI can react.. */
	failedJoinRequest: Subject<any> = new Subject();
}

File: WebSocketChat.Server/WebSocketChat.Server/Models/Chat/ChatRoom.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WebSocketChat.Server.Models.Chat
{
    public class ChatRoom
    {
        public ChatRoom(int roomIdentifier)
        {
            RoomIdentifier = roomIdentifier;
            ConnectedUsers = new List<string>();
        }


        /// <summary>
        /// Identifier of the room. 
        /// </summary>
        public int RoomIdentifier { get; set; }

        /// <summary>
        /// Name of the chatroom
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// Description of the chatroom
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// Base64 string representation of the room's avatar
        /// </summary>
        public string RoomAvatarBase64 { get; set; }

        /// <summary>
        /// Identifier-GUIDs of all users which have joined this room
        /// </summary>
        public List<string> ConnectedUsers { get; set; }

        /// <summary>
        /// Indicates whether this is a private room which is only used for 1v1 whisper-messages
        /// </summary>
        public bool IsPrivateRoom { get; set; }

        /// <summary>
        /// Optional password-protection. This property won't be serialized.
        /// </summary>
        public string Password { get; set; }

        /// <summary>
        /// Indicates whether the room is password protected
        /// </summary>
        public bool IsPasswordProtected => !string.IsNullOrWhiteSpace(Password);

        /// <summary>
        /// The password should never be serialized because it mustn't be send to the user.
        /// This method is only used by Json.NET
        /// </summary>
        public bool ShouldSerializePassword() => false;
    }
}
